pkg/storage/stores/shipper/compactor/retention/retention.go

```diff
@@ -115,30 +115,31 @@ func (t *Marker) markTable(ctx context.Context, tableName string, db *bbolt.DB)
 		return nil
 	})
 	if err != nil {
-		return false, 0, err
+		return false, false, err
 	}
 	if empty {
 		t.markerMetrics.tableProcessedTotal.WithLabelValues(tableName, tableActionDeleted).Inc()
-		return empty, markerWriter.Count(), nil
+		return empty, true, nil
 	}
-	if markerWriter.Count() == 0 {
+	if !modified {
 		t.markerMetrics.tableProcessedTotal.WithLabelValues(tableName, tableActionNone).Inc()
-		return empty, markerWriter.Count(), nil
+		return empty, modified, nil
```

Comment:
tiny nit: but I think it makes it slightly easier to read to hard-code this value to `false`, as it doesn't require the reader to invert the condition (which is already an inversion) that got them here,